frontend/packages/console-app/src/components/guided-tours/QuickStartSection.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { last } from 'lodash';
+import { Link } from 'react-router-dom';
+import { Button } from '@patternfly/react-core';
+import { GuidedTourItem, GuidedTourState, GuidedTourTaskItem } from './utils/guided-tour-typings';
+import { GuidedTourStatus, TaskState } from './utils/guided-tour-status';
+import QuickStartTask from './QuickStartTask';
+import TaskHeader from './TaskHeader';
+import './QuickStartSection.scss';
+
+type QuickStartSectionProps = {
+  quickStart: GuidedTourItem;
+  quickStartState: GuidedTourState;
+  nextCallback?: () => void;
+  reviewCallback?: (reviewState: boolean) => void;
+  prevCallback?: () => void;
+};
+
+enum PrimaryButtonText {
+  START = 'Start Tour',
+  NEXT = 'Next',
+  CLOSE = 'Close',
+}
+
+const QuickStartSection: React.FC<QuickStartSectionProps> = ({
+  quickStart,
+  quickStartState,
+  nextCallback,
+  reviewCallback,
+  prevCallback,
+}) => {
+  const { introduction, tasks, conclusion } = quickStart;
+  const { tourState, tasks: tasksWithStates } = quickStartState;
+
+  const subtitle = (task) =>
+    task.title &&
+    `${tasks.findIndex((taskElement) => taskElement.title === task.title) + 1} of ${tasks.length}`;
+
+  const isFinalScreen: boolean =
+    tourState === GuidedTourStatus.IN_PROGRESS &&
+    tasksWithStates?.length === tasks.length &&
+    (last(tasksWithStates).taskState === TaskState.SUCCESS ||
+      last(tasksWithStates).taskState === TaskState.FAILED);
+
+  const getPrimaryButtonText = (): PrimaryButtonText => {
+    if (isFinalScreen) {
+      return PrimaryButtonText.CLOSE;
+    }
+    if (tourState === GuidedTourStatus.IN_PROGRESS) {
+      return PrimaryButtonText.NEXT;
+    }
+    return PrimaryButtonText.START;
+  };
+
+  const renderTasks = (): GuidedTourTaskItem[] => {
+    if (tourState === GuidedTourStatus.NOT_STARTED) {
+      return [{ description: introduction }];
+    }
+    if (isFinalScreen) {
+      return [...tasksWithStates, ...[{ description: conclusion }]];
+    }
+    return tasksWithStates;
+  };
+
+  return (
+    <>
+      {renderTasks().map((task, index) => (
+        <QuickStartTask
+          title={task?.title}
+          subtitle={subtitle(task)}
+          description={task.description}
+          review={task?.review}
+          reviewState={task.reviewState}
+          reviewCallback={reviewCallback}
+          recapitulation={task?.recapitulation}
+          taskIndex={index + 1}
+          taskState={task?.taskState}
+          isFinalScreen={isFinalScreen}
+          taskHelp={task?.taskHelp}
```

Comment:
Why not just send the whole `task` object and the component can use whatever property it needs to render correctly.